Use new CBOR tags, restructure registry system. (#87)
digitalbazaar/cborld: CHANGELOG.md

# @digitalbazaar/cborld ChangeLog

> ## 8.0.0 - 2024-mm-dd

Reviewer: With no breaking changes, we're targeting 7.1:

```suggestion
## 7.1.0 - 2024-mm-dd
```

### Added
- Add TypedLiteral, UntypedLiteral encoders and decoders to more generally
  handle type-specific behavior such as cryptosuite string encoding as well

>   as pass through uncompressible untyped values.

Reviewer: ```suggestion
  as pass through uncompressible untyped values. Existing workflows that do not
  use new string tables will not result in differently encoded payloads from the
  previous version.
```

> ### CHANGED

Reviewer: ```suggestion
### Changed
```

> - Restructure term registry system to be more general with four tables which

Reviewer: We need a note here that says that, by default, the behavior will be the same as in the previous minor version and that using these parameters is not allowed in `legacy` mode (we should also block these inputs as invalid API use if the `legacy` varint is being used).

  can all be passed by the user during encoding and decoding:

>     - keywordsTable: for JSON-LD keywords (e.g. '@context', '@type')

Reviewer: Why can this be passed by the user? I recommend we don't expose this one at this time.

>     - stringTable: for JSON-LD string values. This subsumes 'appContextMap',

Reviewer: Maybe `appContextMap` should only be allowed in `legacy` mode? And users of the library don't have to know so much about this then. They are either running in legacy mode and use `appContextMap` or they are running in the new mode and they use the new tables instead.

      as context strings can be included here, but includes arbitrary string
      values as well.
    - urlSchemeTable: for URL schemes (i.e. 'http://')
    - typedLiteralTable: for values associated with JSON-LD types. This
      subsumes type-specific codecs like the cryptosuite codec.
- Restructure CBOR-LD tag system to use a range of tags where
  the tag value informs what values for the tables above should be used
  via a registry. Legacy tags (0x0501) are still supported, and new tags are
  in the range 0x0600-0x06FF. In addition, the tag value is part of a varint

>   that begins the CBORLD paylaod if necessary that allows for more registry

Reviewer: ```suggestion
  that begins the CBOR-LD paylaod if necessary that allows for more registry
```
